File: client/src/app/shared/models/professor.ts
import { Telefone } from "./telefone";
import { Endereco } from "./endereco";

export interface Professor {
    id: number;
    nome: string;
    telefone: Telefone;
    endereco: Endereco;
    sexo: string;
    telefoneId: number;
    enderecoId: number;   
}

File: server/edusys.Api/Entities/Professor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace edusys.Api.Entities
{
    [Table("Professor")]

    public class Professor
    {
        [Key]
        public int Id { get; set; }
        public string? Nome { get; set; }
        public DateTime DataNascimento { get; set; }
        public string? Telefone { get; set; }
        public int EnderecoId { get; set; }
        public virtual Endereco? Endereco { get; set; }

    }
}
